src/env.ts

```diff
@@ -8,52 +9,53 @@ export const isProdEnv =
+   * environment. Available values are:
+   * * `default`: Runs the chainhook server and the REST API server (this is the default)
+   * * `writeonly`: Runs only the chainhook server
+   * * `readonly`: Runs only the REST API server
+   */
+  RUN_MODE: Type.Enum(
+    { default: 'default', readonly: 'readonly', writeonly: 'writeonly' },
+    { default: 'default' }
+  ),
 
-export function getEnvVars(): Env {
-  const schema = {
-    type: 'object',
-    required: [
-      'STACKS_API_ENDPOINT',
-      'STACKS_EXPLORER_ENDPOINT',
-      'MEMPOOL_JS_HOSTNAME',
-      'BLOCKCHAIN_INFO_API_ENDPOINT',
-      'BLOCKCHAIN_EXPLORER_ENDPOINT',
-    ],
-    properties: {
-      STACKS_API_ENDPOINT: {
-        type: 'string',
-        default: 'https://stacks-node-api.mainnet.stacks.co',
-      },
-      STACKS_EXPLORER_ENDPOINT: {
-        type: 'string',
-        default: 'https://explorer.stacks.co',
-      },
-      MEMPOOL_JS_HOSTNAME: {
-        type: 'string',
-        default: 'mempool.space',
-      },
-      BLOCKCHAIN_INFO_API_ENDPOINT: {
-        type: 'string',
-        default: 'https://blockchain.info',
-      },
-      BLOCKCHAIN_EXPLORER_ENDPOINT: {
-        type: 'string',
-        default: 'https://www.blockchain.com',
-      },
-    },
-  };
-  const config = envSchema<Env>({
-    schema: schema,
-    dotenv: true,
-  });
-  return config;
-}
+  /** Hostname of the API server */
+  API_HOST: Type.String({ default: '0.0.0.0' }),
+  /** Port in which to serve the API */
+  API_PORT: Type.Number({ default: 3000, minimum: 0, maximum: 65535 }),
+  /** Port in which to receive chainhook events */
+  EVENT_PORT: Type.Number({ default: 3099, minimum: 0, maximum: 65535 }),
+  /** Event server body limit (bytes) */
+  EVENT_SERVER_BODY_LIMIT: Type.Integer({ default: 20971520 }),
+  /** Hostname that will be reported to the chainhook node so it can call us back with events */
+  EXTERNAL_HOSTNAME: Type.String({ default: '127.0.0.1' }),
 
-export const ENV = getEnvVars();
+  /** Hostname of the chainhook node we'll use to register predicates */
+  CHAINHOOK_NODE_RPC_HOST: Type.String({ default: '127.0.0.1' }),
+  /** Control port of the chainhook node */
+  CHAINHOOK_NODE_RPC_PORT: Type.Number({ default: 20456, minimum: 0, maximum: 65535 }),
+  /**
+   * Authorization token that the chainhook node must send with every event to make sure it's
+   * coming from the valid instance
+   */
+  CHAINHOOK_NODE_AUTH_TOKEN: Type.String(),
+
+  PGHOST: Type.String(),
+  PGPORT: Type.Number({ default: 5432, minimum: 0, maximum: 65535 }),
+  PGUSER: Type.String(),
+  PGPASSWORD: Type.String(),
+  PGDATABASE: Type.String(),
+  /** Limit to how many concurrent connections can be created */
+  PG_CONNECTION_POOL_MAX: Type.Number({ default: 10 }),
+  PG_IDLE_TIMEOUT: Type.Number({ default: 30 }),
+  PG_MAX_LIFETIME: Type.Number({ default: 60 }),
+});
+type Env = Static<typeof schema>;
+
+export const ENV = envSchema<Env>({
```

Comment:
This env var setup is 💯 

Would be great to standardize in some of our other products. Also, some of the "schema to documentation" tooling could potentially be used to generate pretty markdown docs for env vars (not that this PR or project needs that, just thinking out loud).